diff --git a/node_modules/@antv/g2/lib/geometry/label/util/createWorker.js b/node_modules/@antv/g2/lib/geometry/label/util/createWorker.js
index 7b29f33..e11b9f3 100644
--- a/node_modules/@antv/g2/lib/geometry/label/util/createWorker.js
+++ b/node_modules/@antv/g2/lib/geometry/label/util/createWorker.js
@@ -8,6 +8,9 @@ function createWorker(f) {
     }
     catch (e) {
         // @ts-ignore
+        if (typeof window === 'undefined') {
+            return;
+        }
         blob = new window.BlobBuilder();
         blob.append(f.toString());
         blob = blob.getBlob();
